python_modules/dagster/dagster/_core/definitions/asset_graph.py

```diff
@@ -301,3 +303,43 @@ def asset_check_keys(self) -> AbstractSet[AssetCheckKey]:
             *(key for ad in self.assets_defs for key in ad.check_keys),
             *(key for acd in self.asset_checks_defs for key in acd.keys),
         }
+
+    def get_subset(
+        self,
+        executable_asset_keys: AbstractSet[AssetKey],
+        asset_check_keys: Optional[AbstractSet[AssetCheckKey]] = None,
+    ) -> "AssetGraph":
+        """Returns a new asset graph where only the provided asset keys are executable. All parent
+        keys of a selected executabel asset will be included as unexecutable external assets (unless
+        they are themselves present in the executable selection).
+        """
+        from dagster._core.definitions.external_asset import (
+            create_unexecutable_external_assets_from_assets_def,
+        )
+
+        invalid_executable_keys = executable_asset_keys - self.executable_asset_keys
+        check.invariant(
+            not invalid_executable_keys,
+            "Provided executable asset keys must be a subset of existing executable asset keys."
+            f" Invalid provided keys: {invalid_executable_keys}",
+        )
+
+        executable_assets_defs, raw_loadable_assets_defs = subset_assets_defs(
+            self.assets_defs, executable_asset_keys, asset_check_keys
+        )
+        loadable_assets_defs = [
+            unexecutable_ad
+            for ad in raw_loadable_assets_defs
+            for unexecutable_ad in create_unexecutable_external_assets_from_assets_def(ad)
+        ]
+
+        # ignore check keys that don't correspond to an AssetChecksDefinition
+        asset_checks_defs = list(
+            {
+                acd
+                for key, acd in self._asset_check_compute_defs_by_key.items()
+                if key in (asset_check_keys or []) and isinstance(acd, AssetChecksDefinition)
+            }
+        )
```

Comment:
we should probably wait for your diff to kill `AssetChecksDefinition`

Reply:
that's actually stacked on this one so it's easier to land this first